app/components/candidate_interface/gcse_equivalency_required_component.rb

```diff
@@ -0,0 +1,39 @@
+class CandidateInterface::GcseEquivalencyRequiredComponent < ViewComponent::Base
+  attr_accessor :application_choice, :missing_uk_gcses, :accept_english_gcse_equivalency, :accept_maths_gcse_equivalency, :accept_science_gcse_equivalency
+
+  def initialize(application_choice, missing_uk_gcses)
+    @application_choice = application_choice
+    @missing_uk_gcses = missing_uk_gcses
+    @accept_english_gcse_equivalency = application_choice.course.accept_english_gcse_equivalency?
+    @accept_maths_gcse_equivalency = application_choice.course.accept_maths_gcse_equivalency?
+    @accept_science_gcse_equivalency = application_choice.course.accept_science_gcse_equivalency?
+  end
+
+  def provider_accepts_equivalencies_and_gcse_is_missing?
+    missing_gcse_subjects = missing_uk_gcses.map(&:subject)
+
+    accepted_equivalency_subjects = []
+    accepted_equivalency_subjects << 'english' if accept_english_gcse_equivalency
+    accepted_equivalency_subjects << 'maths' if accept_maths_gcse_equivalency
+    accepted_equivalency_subjects << 'science' if accept_science_gcse_equivalency
+
+    missing_gcse_subjects == accepted_equivalency_subjects
+  end
```

Comment:
I discovered a slight logic flaw here when testing it locally.

If the provider accepts equivalencies in English + maths, but the candidate is only missing one of those, then then method returns false and the component shows the blue inset text warning when it should be showing just a sentence.

I think instead the logic should be something like

```ruby
missing_gcse_subjects.all? {|missing_subject|  accepted_equivalency_subjects.include?(missing_subject) }
```

...ie check that the list of accepted equivalency subjects contains all the subjects the candidate is missing.

Reply:
Good spot!

Reply:
Awesome thanks Frankie, I've implemented your logic and added a unit test to cover that specific scenario 👍